### ISSUE TITLE
`cargo expand` hangs in an infinite loop

### ISSUE THREAD

**Issue by chabapok** [dtolnay/cargo-expand#161]
I try to expand that program 

```
extern crate pest;
#[macro_use]
extern crate pest_derive;

#[derive(Parser)]
#[grammar = "grm.pest"]
pub struct MyParser;

fn main() { }
```

The `grm.pest`  file:

```
A = { "A" }
B = { "B" }
file = {
    SOI ~
    (( A | B )? ~ NEWLINE)* ~
    EOI
}
```

And `cargo expand` command hangs in an infinite loop.

**Comment by ehuss** [comment]
This looks to be an issue in rustfmt, I have posted https://github.com/rust-lang/rustfmt/issues/5500 to track that issue.

It should eventually finish, or you can use `--ugly` to skip formatting.

**Comment by dtolnay** [comment]
The `prettyplease` formatter formats this in 0.4 milliseconds. You could try reinstalling with `cargo install cargo-expand --force --features prettyplease`.

At some point we should make that the default behavior.

**Comment by ehuss** [comment]
> At some point we should make that the default behavior.

Are there any known blockers?

**Comment by dtolnay** [comment]
Not really, I have been using it since it was added. I released 1.0.30 with prettyplease formatting by default.
